### PR TITLE
LG-3330: password input controlled value

### PR DESCRIPTION
**Pull request by shaneeza**
## ✍️ Proposed changes

🎟 _Jira ticket:_ [LG-3330](https://jira.mongodb.org/browse/LG-3330)

### For bug fixes, new features & breaking changes

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

### For new components

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README
- [ ] I have verified the Live Example looks as intended on the design website.

### CODE REVIEW

File: packages/password-input/src/PasswordInput.story.tsx
Comment (shaneeza) [inline packages/password-input/src/PasswordInput.story.tsx:1]: revert these changes

File: .changeset/late-ants-remain.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@leafygreen-ui/hooks': patch
```
Comment (shaneeza) [inline .changeset/late-ants-remain.md:2]: Should be major since, im changing what is returned

File: packages/hooks/src/useControlledValue/useControlledValue.spec.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ describe('packages/lib/useControlledValue', () => {
 
     act(() => {
       current.handleChange({ target: { value: 'banana' } } as ChangeEvent<any>);
-      current.setInternalValue('banana');
+      current.setUncontrolledValue('banana');
```
Comment (TheSonOfThomp) [inline packages/hooks/src/useControlledValue/useControlledValue.spec.ts:19]: We should add more tests to this. 
The current one should be named e.g. "`handleChange` calls provided callback".

We should create another test where we update the value passed into the hook and expect that the current.value changes

**Review by TheSonOfThomp — APPROVED** [review]
